### PR TITLE
separate setting and processing data in generate_road_colours.py

### PR DESCRIPTION
**Pull request by matkoniecz**
It intends to make this script more readable and to make value tweaking easier.

### CODE REVIEW

File: scripts/generate_road_colours.py [hunk 1]
```diff
@@ -86,11 +91,12 @@ def rgb_error(self):
...
+for road in colourless_roads:
+  shield_colours[road] = Color((shield_colour_info.end_l, 0, 0))
 
 print "\n\nRoad shield information\n\n"
 for name, colour in shield_colours.iteritems():
   print "{name}:{rgb}".format(name = name, rgb = colour.rgb())
 
 for name, colour in shield_colours.iteritems():
-  print "# {name}: {lch}, error {delta:.1f}".format(name = name, lch = colour.lch(), delta = colour.rgb_error())
+  print "# {name}: {lch}, error {delta:.1f}".format(name = name, lch = colour.lch(), delta = colour.rgb_error())
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: there should be a newline at the end of the file

Comment (matkoniecz) [inline scripts/generate_road_colours.py]: OK

**Comment by math1985** [comment]
@pnorman You can probably review this faster than I.

File: scripts/generate_road_colours.py [hunk 2]
```diff
@@ -1,47 +1,17 @@
 #!/usr/bin/env python2
-
 from colormath.color_conversions import convert_color
 from colormath.color_objects import LabColor, LCHabColor, sRGBColor
 from colormath.color_diff import delta_e_cie2000
 import numpy
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: We don't need to import numpy anymore

File: scripts/generate_road_colours.py [hunk 3]
```diff
@@ -54,9 +24,45 @@ def rgb_error(self):
 line_colour_infos["casing"] = ColourInfo(start_l = 50, end_l = 50, start_c = 70, end_c = 55)
 line_colour_infos["low-zoom-casing"] = ColourInfo(start_l = 50, end_l = 70, start_c = 50, end_c = 65)
 
+shield_colour_info = ColourInfo(start_l = 65, end_l = 80, start_c = 30, end_c = 30)
+colourless_roads = ["teriary"]
+
+
+
+
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: Keep the gap to one blank line, and if needed explain what's below in a comment

File: scripts/generate_road_colours.py [hunk 4]
```diff
@@ -54,9 +24,45 @@ def rgb_error(self):
...
+#Start processing set data
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: Space after `#`

File: scripts/generate_road_colours.py [hunk 5]
```diff
@@ -54,9 +24,45 @@ def rgb_error(self):
...
+
+h = min_h
+for name in road_classes:
+  hues[name] = h
+  h = (h + delta_h) % 360
+
 # Colours for the MSS
 colours = OrderedDict()
 
+class Color:
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: Classes are normally at the top of a file, or in their own file. If it was worth it, we'd do the latter, but that doesn't seem the best for what is a style project, not a code project.

File: scripts/generate_road_colours.py [hunk 6]
```diff
@@ -1,47 +1,17 @@
...
-    return delta_e_cie2000(convert_color(self.m_lch, LabColor),
-                           convert_color(sRGBColor.new_from_rgb_hex(self.rgb()), LabColor))
 
 road_classes = ["motorway", "trunk", "primary", "secondary"]
-colour_divisions = len(road_classes) - 1
-hues = OrderedDict()
 
 # The minimum and maximum hue for the road colours
-# Because hue is a circle, it may be needed to add/subtract 360 to the min or
-# max when changing them
```
Comment (pnorman) [inline scripts/generate_road_colours.py]: This comment should go with where `min_h` and `max_h` is set. Unless it's not true - I never did test to see if it was necessary. To get between two colours on a circle you can go the long way or the short way.

**Comment by pnorman** [comment]
@matkoniecz ping

**Comment by matkoniecz** [comment]
@pnorman I am alive and improving this PR is on my TODO list  - unfortunately at this moment I am unable to promise that I will do it in a short time.

**Comment by math1985** [comment]
@matkoniecz Just wondering if you will have time to look at this at some point?

**Comment by math1985** [comment]
I'll close this in the meanwhile, feel free to re-open when ready!

**Pull request closed without merging**